### ISSUE TITLE
Linker warnings on latest mac 10.15 beta

### ISSUE THREAD

**Issue by malkomalko** [crystal-lang/crystal#7884]
Preface: These are just annoying warnings, but crystal still works/builds on 10.15, however it makes running specs hard to parse the output without redirecting errors, which is pretty bad for testing :D.

```
>  crystal -v
Crystal 0.29.0 (2019-06-10)

LLVM: 6.0.1
Default target: x86_64-apple-macosx
```

On the latest mac beta (10.15 Beta 19A471t), `ld` is throwing a bunch of warnings.  I've done a bunch of searching and can't find much information on this.  A typical trace will look something like this:

```
ld: warning: building for macOS, but linking in object file (D-ir.o) built for
ld: warning: building for macOS, but linking in object file (_main.o) built for
ld: warning: building for macOS, but linking in object file (C-rystal5858S-ystem5858D-ir.o) built for
ld: warning: building for macOS, but linking in object file (E-rrno.o) built for
ld: warning: building for macOS, but linking in object file (S-tring.o) built for
ld: warning: building for macOS, but linking in object file (A-rgumentE-rror.o) built for
ld: warning: building for macOS, but linking in object file (C-allS-tack.o) built for
ld: warning: building for macOS, but linking in object file (A-rray40P-ointer40V-oid4141.o) built for
ld: warning: building for macOS, but linking in object file (P-ointer40P-ointer40V-oid4141.o) built for
ld: warning: building for macOS, but linking in object file (E-xception.o) built for
ld: warning: building for macOS, but linking in object file (P-ointer40L-ibU-nwind5858E-xception41.o) built for
ld: warning: building for macOS, but linking in object file (U-I-nt64.o) built for
ld: warning: building for macOS, but linking in object file (G-C-.o) built for
ld: warning: building for macOS, but linking in object file (P-ointer40U-I-nt841.o) built for
ld: warning: building for macOS, but linking in object file (S-tring5858B-uilder.o) built for
ld: warning: building for macOS, but linking in object file (S-lice40T-41.o) built for
ld: warning: building for macOS, but linking in object file (S-lice40U-I-nt841.o) built for
ld: warning: building for macOS, but linking in object file (I-O-5858E-ncoder.o) built for
ld: warning: building for macOS, but linking in object file (I-conv.o) built for
ld: warning: building for macOS, but linking in object file (M-ath.o) built for
ld: warning: building for macOS, but linking in object file (I-nt32.o) built for
ld: warning: building for macOS, but linking in object file (S-taticA-rray40U-I-nt84432102441.o) built for
ld: warning: building for macOS, but linking in object file (I-ndexE-rror.o) built for
...
```

**Comment by asterite** [comment]
I don't think there's much we can do until the new Mac version releases. These issues usually resolve themselves or it's easier to tackle them once the Mac version is out (for instance, I wouldn't upgrade my computer to a beta version to try to fix this issue, and probably not many will).

**Comment by malkomalko** [comment]
Perfectly valid response, and like I said it's not blocking, just an inconvenience that I'm more than happy to live with.

**Comment by jacob-carlborg** [comment]
Not sure if you have already fixed this but have a look at: https://github.com/golang/go/issues/33214#issuecomment-541353859.

**Comment by RX14** [comment]
Can someone try building again using `--target x86_64-apple-macosx10.4.0` and see if it changes anything?
